sdk/appcenter-identity/src/androidTest/java/com/microsoft/appcenter/identity/storage/PreferenceTokenStorageAndroidTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.identity.storage;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+
+import com.microsoft.appcenter.utils.UUIDUtils;
+import com.microsoft.appcenter.utils.context.AuthTokenInfo;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+
+import static com.microsoft.appcenter.identity.storage.PreferenceTokenStorage.TOKEN_HISTORY_LIMIT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class PreferenceTokenStorageAndroidTest {
+
+    private static final String AUTH_TOKEN = UUIDUtils.randomUUID().toString();
+
+    private static final String ACCOUNT_ID = UUIDUtils.randomUUID().toString();
+
+    private Context mContext;
+
+    @Before
+    public void setUp() {
+        mContext = InstrumentationRegistry.getTargetContext();
+        SharedPreferencesManager.initialize(mContext);
+    }
+
+    @After
+    public void tearDown() {
+        SharedPreferencesManager.clear();
+    }
+
+    @Test
+    public void saveToken() {
+        PreferenceTokenStorage tokenStorage = new PreferenceTokenStorage(mContext);
+
+        /* Save the token into storage. */
+        tokenStorage.saveToken(AUTH_TOKEN, ACCOUNT_ID, new Date());
+
+        /* Assert that storage returns the same token. */
+        assertEquals(AUTH_TOKEN, tokenStorage.getToken());
+        assertEquals(ACCOUNT_ID, tokenStorage.getHomeAccountId());
+
+        /* Remove the token from storage. */
+        tokenStorage.saveToken(null, null, null);
+
+        /* Assert that there's no token in storage. */
+        assertNull(tokenStorage.getToken());
+        assertNull(tokenStorage.getHomeAccountId());
+
+        /* The same token should't be in history twice in a row. */
+        tokenStorage.saveToken(AUTH_TOKEN, ACCOUNT_ID, new Date());
+        tokenStorage.saveToken(AUTH_TOKEN, ACCOUNT_ID, new Date());
```

Comment:
We should use 2 fixed different dates as currently those 2 dates will likely be the same millisecond and thus equal so not really proving the point that we are trying to test (even if dates different if same value we don't store twice).